static/app/components/events/interfaces/spans/utils.tsx

```diff
@@ -545,32 +552,52 @@ export function getMeasurements(event: EventTransaction): Map<number, VerticalMa
     const name = measurement.name.slice('mark.'.length);
     const value = measurement.value;
 
-    if (mergedMeasurements.has(measurement.timestamp)) {
-      const verticalMark = mergedMeasurements.get(measurement.timestamp) as VerticalMark;
+    const bounds = generateBounds({
+      startTimestamp: measurement.timestamp,
+      endTimestamp: measurement.timestamp,
+    });
 
-      verticalMark.marks = {
-        ...verticalMark.marks,
-        [name]: value,
-      };
+    // This condition should never be hit
+    if (bounds.type !== 'TIMESTAMPS_EQUAL') {
+      return;
+    }
```

Comment:
I think the comment should probably explain `why` it shouldn't be hit vs. that it shouldn't be, it's not really clear to me without going and looking as to what generate bounds is doing. (what vs. why style comment)

Reply:
Agreed, I made the assumption that it would be self-explanatory because it's using the same timestamp value for start and end in the call to `generateBounds` above, I'll add that onto the comment 👍